File: packages/react-native-nitro-test-external/ios/HybridTestView.swift
//
//  HybridTestView.swift
//  react-native-nitro-test
//
//  Created by Marc Rousavy on 03.10.24.
//

import NitroModules
import UIKit

class HybridTestView: HybridTestViewSpec {
  // UIView
  var view: UIView = UIView()

  // Props
  var isCyan: Bool = false {
    didSet {
      view.backgroundColor = isCyan ? .systemCyan : .systemBlue
    }
  }
  var hasBeenCalled: Bool = false
  var testCallback: () -> Void = {}

  // Methods
  func testMethod() throws {
    hasBeenCalled = true
    testCallback()
  }
}

File: packages/react-native-nitro-test-external/android/src/main/java/com/margelo/nitro/test/external/HybridTestView.kt
package com.margelo.nitro.test.external

import android.graphics.Color
import android.view.View
import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip
import com.facebook.react.uimanager.ThemedReactContext

@Keep
@DoNotStrip
class HybridTestView(
  val context: ThemedReactContext,
) : HybridTestViewSpec() {
  // View
  override val view: View = View(context)

  // Props
  override var isCyan: Boolean = false
    set(value) {
      field = value
      val color = if (value) Color.CYAN else Color.BLUE
      view.setBackgroundColor(color)
    }
  override var hasBeenCalled: Boolean = false
  override var testCallback: () -> Unit = {}

  // Methods
  override fun testMethod() {
    hasBeenCalled = true
    testCallback()
  }
}
